app/controllers/users_controller.rb

```diff
@@ -691,6 +703,18 @@ def create
       user.custom_fields = fields
     end
 
+    # Handle associated accounts
+    associations = []
+    if params[:external_ids]&.is_a?(ActionController::Parameters) && current_user&.admin? && is_api?
+      params[:external_ids].each do |provider_name, provider_uid|
+        authenticator = Discourse.enabled_authenticators.find { |a| a.name == provider_name }
+        raise Discourse::InvalidParameters.new(:external_ids) if !authenticator&.is_managed?
+
+        association = UserAssociatedAccount.find_or_initialize_by(provider_name: provider_name, provider_uid: provider_uid)
```

Comment:
Is it possible that this will find an associated account from a different user? I guess it's fine since the uid is unique?

Reply:
It can find an associated account that was previously associated to another account, but that is fine because that association will be moved to current user. This is possible anyway if user de-associates their account and then authenticated with the new one.

I had same concerns and then I noticed we have similar logic in [core](https://github.com/discourse/discourse/blob/main/lib/auth/managed_authenticator.rb#L52-L54).